packages/material/scss/button/_theme.scss

```diff
@@ -23,13 +25,23 @@
         @include appearance( button );
         box-shadow: $button-shadow;
         font-weight: 500;
-        transition: box-shadow 280ms cubic-bezier(.4, 0, .2, 1);
+        transition: box-shadow 280ms cubic-bezier(.4, 0, .2, 1), background-color 280ms cubic-bezier(.4, 0, .2, 1);
+
+        // Hovered state
+        &:hover,
+        &.k-state-hover {
+            @include appearance( hovered-button );
+        }
 
         // Focused state
-        &:not(.k-no-focus):focus,
-        &:not(.k-no-focus).k-state-focused {
-            background-color: $button-focused-bg;
+        &:focus,
+        &.k-state-focused {
             @include appearance( focused-button );
+            background-color: $button-focused-bg;
+        }
+        &.k-no-focus {
+            @include appearance( button );
+            box-shadow: $button-shadow;
```

Comment:
I think that the ripple effect should hide the focused background, too (which comes from the `:focus` class on line 40). This is how Angular Material behaves, but that leaves the button without a focused state when the ripple effect finishes. Thoughts?